Kinvey.Core/Store/MultiInsertRequest.cs

```diff
@@ -244,23 +242,65 @@ private async Task<KinveyMultiInsertResponse<T>> HandleNetworkRequestAsync(IList
                 }
             }
 
-            var multiInsertKinveyDataStoreResponse = new KinveyMultiInsertResponse<T>
+            var multiInsertNetworkResponse = new KinveyMultiInsertResponse<T>
             {
                 Entities = new List<T>(),
                 Errors = new List<Error>()
             };
 
             if (entitiesToMultiInsert.Count > 0)
             {
-                var multiInsertRequest = Client.NetworkFactory.BuildMultiInsertRequest<T, KinveyMultiInsertResponse<T>>(Collection, entitiesToMultiInsert);
-                multiInsertKinveyDataStoreResponse = await multiInsertRequest.ExecuteAsync();
+                var countOfMultiInsertOperations = Math.Ceiling(entitiesToMultiInsert.Count / (double)Constants.NUMBER_LIMIT_OF_ENTITIES);
+                var currentIndex = 0;
+                var currentCountOfMultiInsertOperations = 0;
+
+                while (currentCountOfMultiInsertOperations < countOfMultiInsertOperations)
+                {
+                    var tasks = new List<Task<KinveyMultiInsertResponse<T>>>();
+
+                    for (var index = currentCountOfMultiInsertOperations; index < currentCountOfMultiInsertOperations + 10; index++)
+                    {
+                        if (index < countOfMultiInsertOperations)
+                        {
+                            tasks.Add(HandleMultiInsertRequestAsync(entitiesToMultiInsert.Skip(index * Constants.NUMBER_LIMIT_OF_ENTITIES).Take(Constants.NUMBER_LIMIT_OF_ENTITIES).ToList()));
+                        }
+                    }
+
+                    await Task.WhenAll(tasks.ToArray());
```

Comment:
I'm wondering if this could result in a big memory consumption. Maybe better to perform `await` in chunks. Maybe in chunks of `X` number of processors available in the device.

Reply:
@vinaygahlawat  @heyzooi 

The matter is we are performing it in chunks. If you have a look at the algorithm above (line 261), you can see that there are only 10 threads in one time. Our algorithm is following:
1.	To learn the maximum count of threads. 
For example.
2001 – count of entities, 100 – count of entities in one request.
2001/100 = 21 – count of requests (count of threads)
2.	To send these requests by chunks. Every chunk will have 10 requests(threads.)
According to our example we will have 3 iterations:
a)	Sending 10 requests in 10 parallel threads. Waiting for their finishing.
b)	Sending the next 10 requests in 10 parallel threads. Waiting for their finishing.
c)	Sending the last 1 request in 1 thread. Waiting for its finishing.
Talk about 10. I took this constant from the push logic. It was implemented long time ago before my joining to the project. It means it has been working for a long period without any troubles.  At least I hope for it))

@heyzooi About your suggestion of changing threads count in chunk depending on processors available in the device. I think it is a good idea. It will help improve performance. For example, now if we launch this code in a device with 2 processors, we will have concurrent access of 10 threads to these 2 processors. As a result, switching over a thread context will happen. It will impact performance.
@Vinay If we are going to implement Victor’s suggestion, I think it is better to do in a separate task as we need to make the same changes in push logic too. 
 
On the other hand, we discussed with @vinaygahlawat  the issue of performance before starting the task.
Basically a developer can change the count of entities to be saved in his implementation, if he sees about performance issue for some count of entities. For instance, he can simply decrease a limit of entities which he is going to send to our multi insert API method.

Reply:
@heyzooi @eugeninnowise I do not think that considering the changing of thread count dynamically is in the scope of this ticket. Let's have a separate discussion about the benefits and costs of implementing that, but it should not be blocking the approval of this PR.